capstone/src/main/webapp/js/autocomplete.js

```diff
@@ -0,0 +1,166 @@
+ */
+function addActiveTag(autocompleteListElmt) {
+  if (!autocompleteListElmt) {
+    return;
+  }
+  removeActiveTag(autocompleteListElmt);
+  if (selectedSuggestionPosition >= autocompleteListElmt.length) {
+    selectedSuggestionPosition = 0;
+  } else if (selectedSuggestionPosition < 0) {
+    selectedSuggestionPosition = (autocompleteListElmt.length - 1);
+  }
+  autocompleteListElmt[selectedSuggestionPosition].classList.add(ACTIVE_CLASS);
+}
+
+/**
+ * Removes the 'active' tag from an autocomplete elmt
+ * @param {Object} autocompleteListElmt
+ */
+function removeActiveTag(autocompleteListElmt) {
+  for (elmt of autocompleteListElmt) {
+    elmt.classList.remove(ACTIVE_CLASS);
+  }
+}
+
+/**
+ * Closes dropdown autocomplete list
+ * Does not close the element provided
+ * @param {Object} elmnt
+ */
+function closeAllElmntExcept(elmnt) {
+  const autoItems = document.getElementsByClassName(ITEM_CLASS);
+  for (const item of autoItems) {
+    if (item != elmnt && item != SEARCH_INPUT) {
+      item.parentNode.removeChild(item);
+    }
+  }
+}
+
+/**
+ * Adds valid suggestions to list of stored suggestions
+ * @param{string} validSuggestion
+ */
+function keepTrackOfSuggestions(validSuggestion) {
+  const parent = document.getElementById('suggestions');
+  const listElt = document.createElement('li');
+  listElt.innerHTML = validSuggestion;
+  parent.appendChild(listElt);
+}
+
+/**
+ * Current placeholder until servlet is created
+ * @return{Array<string>} arr
+ */
+function getListOfSuggestions() {
+  const arr = ['Hallo', 'Hello', 'Hi', 'Hiya', 'Howdy', 'Wassup'];
+  return arr;
+}
+
+/**
+ * Identifies and returns relevant suggestions in the arr
+ * @param {string[]} arr
+ * @param {string} val
+ * @return {string[]} relevantSuggestions
+ */
+function getRelevantDataSuggestions(arr, val) {
+  const relevantSuggestions = [];
+  val = val.toUpperCase();
+  for (const arrElt of arr) {
+    if (arrElt.substr(0, val.length).toUpperCase() === val.toUpperCase()) {
+      relevantSuggestions.push(arrElt);
+    }
+  }
+  return relevantSuggestions;
+}
+
+/**
+ * Takes relevant suggestions and displays them in DOM
+ * @param {Object} relevantSuggestions
+ * @param {Object} autocompleteList
+ * @param {string} val
+ */
+function displaySuggestions(relevantSuggestions, autocompleteList, val) {
+  for (arrElt of relevantSuggestions) {
+    const listElmt = document.createElement('div');
+    listElmt.innerHTML = '<strong>' +
+                         arrElt.substr(0, val.length) +
+                         '</strong>';
+    listElmt.innerHTML += arrElt.substr(val.length);
+    const cpyArrElt = arrElt;
+    listElmt.addEventListener('click', () => {
+      SEARCH_INPUT.value = cpyArrElt;
+      keepTrackOfSuggestions(cpyArrElt);
+      closeAllElmntExcept();
+    });
+    autocompleteList.appendChild(listElmt);
+  }
+}
+
+/**
+ * Event occurrance when input is provided to autocomplete field
+ */
+function giveSuggestions() {
```

Comment:
nit: in term of method naming, giveSuggestions sounds like it's going to give the caller (return to caller) new suggestions. But in reality it's getting suggestions and populating them on UI. So something like {display,populate,show}Suggestions might read better. That being said you already have a `displaySuggestions` method that is doing that actual displaying. So this method feels like pre-work.

Here are few other name suggestions: triggerAutocomplete, refreshSuggestions, etc.

Alternatively you can rename the current displaySuggestions method to something else like populateSuggestionsInList. Then rename this method to displaySuggestions.

Reply:
I renamed the giveSuggestions to getAllProposedSuggestions-> how does that work?